tests/realm.cpp

```diff
@@ -471,3 +474,67 @@ TEST_CASE("ShareRealm: in-memory mode from buffer") {
         REQUIRE_THROWS(Realm::get_shared_realm(config3)); // both buffer and encryption
     }
 }
+
+TEST_CASE("ShareRealm: realm closed in did_change callback") {
+    TestFile config;
+    config.schema_version = 1;
+    config.schema = Schema{
+        {"object", {
+            {"value", PropertyType::Int, "", "", false, false, false}
+        }},
+    };
+    config.cache = false;
+    config.automatic_change_notifications = false;
+    auto r1 = Realm::get_shared_realm(config);
+
+    r1->begin_transaction();
+    auto table = r1->read_group().get_table("class_object");
+    auto row_idx = table->add_empty_row(1);
+    auto table_idx = table->get_index_in_group();
+    r1->commit_transaction();
+
+    // Cannot be a member var of Context since Realm.close will free the context.
+    static SharedRealm* shared_realm;
+    shared_realm = &r1;
+    struct Context : public BindingContext {
+        void did_change(std::vector<ObserverState> const&, std::vector<void*> const&, bool) override
+        {
+            (*shared_realm)->close();
+            (*shared_realm).reset();
+        }
+    };
+
+    SECTION("did_change") {
+        r1->m_binding_context.reset(new Context());
+        r1->invalidate();
```

Comment:
To make the `m_group` null so it can run with the below `else` routine in `notify()`:

```c++
    if (m_auto_refresh) {
        if (m_group) {
            m_coordinator->advance_to_ready(*this);
        }
        else  {
            if (m_binding_context) {
                m_binding_context->did_change({}, {});
            }
            if (!is_closed()) {
                m_coordinator->process_available_async(*this);
            }
        }
    }

```